src/libvirtApi/domain.js

```diff
@@ -1089,3 +1089,29 @@ export async function domainAddTPM({ connectionName, vmName }) {
     const args = ["virt-xml", "-c", `qemu:///${connectionName}`, "--add-device", "--tpm", "default", vmName];
     return cockpit.spawn(args, { err: "message", superuser: connectionName === "system" ? "try" : null });
 }
+
+export function domainAttachVnc({ connectionName, vmName, vncAddress, vncPort, vncPassword }) {
```

Comment:
The two are almost identical, and so are the dialogs. Merging the dialogs is a bit more work, but the  API can easily be merged by adding either an `edit` boolean or a `mode` string that supports "add" or "edit" (and throw on any other value).

Reply:
or using `consoleDetail == null` to detect the "add" mode. I'll do that.